docs/samples/models/ibm-granite-2b-gpu/model.yaml

```diff
@@ -0,0 +1,103 @@
+  replicas: 1
+  router:
+    route: { }
+    # Connect to MaaS-enabled gateway
+    gateway:
+      refs:
+        - name: maas-default-gateway
+          namespace: openshift-ingress
+  template:
+    # Ensure the pod is scheduled on GPU nodes
+    nodeSelector:
+      nvidia.com/gpu.present: "true"
+    tolerations:
+      - effect: NoSchedule
+        key: nvidia.com/gpu
+        operator: Exists
+    containers:
+      - name: main
+        image: vllm/vllm-openai:v0.6.3.post1
+        command: ["python3", "-m", "vllm.entrypoints.openai.api_server"]
+        args:
+          - "--model=/mnt/models"
+          # Model name used in API calls (e.g., in the "model" field of requests)
+          # Uses the model name from spec.model.name
+          - "--served-model-name={{ .Spec.Model.Name }}"
+          - "--max-model-len=4096"
+          - "--trust-remote-code"
+          - "--ssl-keyfile=/var/run/kserve/tls/tls.key"
+          - "--ssl-certfile=/var/run/kserve/tls/tls.crt"
+          - "--port=8000"
+          - "--dtype=float16"
+        env:
+          - name: VLLM_LOGGING_LEVEL
+            value: INFO
+          - name: HF_HUB_CACHE
+            value: /models-cache
+          - name: VLLM_ATTENTION_BACKEND
+            value: XFORMERS
+          - name: VLLM_USE_V1
+            value: "false"
+        volumeMounts:
+          - name: cache
+            mountPath: /models-cache
+          - name: tls-certs
+            mountPath: /var/run/kserve/tls
+            readOnly: true
+        resources:
+          limits:
+            cpu: '2'
+            memory: 12Gi
+            nvidia.com/gpu: '1'
+          requests:
+            cpu: '500m'
+            memory: 8Gi
+            nvidia.com/gpu: '1'
+        livenessProbe:
+          httpGet:
+            path: /health
+            port: 8000
+            scheme: HTTPS
+          initialDelaySeconds: 120
+          periodSeconds: 30
+          timeoutSeconds: 30
+          failureThreshold: 5
+        readinessProbe:
+          httpGet:
+            path: /health
+            port: 8000
+            scheme: HTTPS
+          initialDelaySeconds: 120
+          periodSeconds: 10
+          timeoutSeconds: 10
+          failureThreshold: 30
+    volumes:
+      - name: cache
+        emptyDir: {}
+      - name: tls-certs
+        secret:
+          secretName: kserve-tls
+  # Autoscaling can be configured based on your monitoring setup
+  # Uncomment and adjust the following section if you have GPU metrics available
+  # autoscaler:
+  #   metrics:
+  #     # Example: Custom GPU utilization metric (requires GPU monitoring like NVIDIA DCGM)
+  #     - type: Pod
+  #       pod:
+  #         metric:
+  #           name: gpu-utilization  # Replace with your actual metric name
+  #         target:
+  #           type: AverageValue
+  #           averageValue: "80"  # Scale when GPU utilization exceeds 80%
+  #   targetRequests: 100
+  #   scaleUpRate: 10
```

Comment:
There is no `autoscaler` field in `LLMInferenceService`. I'm not sure why this is here.